sonar-php-plugin/src/test/java/org/sonar/plugins/php/reports/phpunit/PhpUnitSensorTest.java

```diff
@@ -113,6 +118,56 @@ public void shouldUseMultiPathCoverage() throws IOException {
       .contains(resourcesFolder + "', nothing will be imported from this report.");
   }
 
+  @Test
+  public void shouldLogWarningWhenTestCaseIsDetectedWithoutDeclaration() {
+    Set.of("src/App.php", "src/EmailTest.php").forEach(
+      file -> context.fileSystem().add(inputFile(file))
+    );
+
+    Sensor sensor = createSensor();
+    sensor.execute(context);
+
+    assertThat(logTester.logs(LoggerLevel.WARN)).contains(EXPECTED_MESSAGE);
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).hasSize(1);
+    assertThat(logTester.logs(LoggerLevel.DEBUG).get(0)).startsWith("Detected and undeclared test case in").endsWith("src/EmailTest.php");
```

Comment:
Why not:
```suggestion
    assertThat(logTester.logs(LoggerLevel.DEBUG).get(0))
      .isEqualTo("Detected and undeclared test case in src/EmailTest.php");
```
?
2. I would suggest exchanging lines 131 & 132. Now if the test fails, I don't know what is in log message.

Reply:
This will fail because the file URI is used which will be an absolute file path.